File: Seccion 11 - Funciones en C++/Seccion_11_Python/19_ejercicio_15.py

nfilas = 0
ncols = 0
matriz = []

def pedir_datos():
    global nfilas, ncols, matriz
    
    nfilas = int(input('Digite el número de filas: '))
    ncols = int(input('Digite el número de columnas: '))
    
    for i in range(nfilas):
        fila = []
        for j in range(ncols):
            val = int(input('[{}][{}] Digite un número: '.format(i, j)))
            fila.append(val)
        matriz.append(fila)

def comprobar_simetria(m):
    cont = 0
    
    if nfilas == ncols:
        for i in range(nfilas):
            for j in range(ncols):
                if m[i][j] == m[j][i]:
                    cont += 1
    
    if cont == nfilas * ncols:
        print('\nLa matriz es Simétrica!\n')
    else:
        print('\nLa matriz NO es Simétrica!\n')

pedir_datos()
comprobar_simetria(matriz)
        

File: Seccion 11 - Funciones en C++/Seccion_11_Dart/19_ejercicio_15.dart

import 'dart:io';

class Matriz {
  int nfilas = 0;
  int ncols = 0;
  List<List> matriz = new List<List>();

  Matriz(this.nfilas, this.ncols) {
    for (int i = 0; i < nfilas; i++) {
      matriz.insert(i, new List());
      for (int j = 0; j < ncols; j++) {
        matriz[i].insert(j, 0);
      }
    }
  }
}

int nfilas, ncols;
List<List> matriz = new List<List>();

void pedirDatos(List<List> m) {
  int val;
  List<int> fila = new List<int>();

  stdout.writeln('Digite el número de filas: ');
  nfilas = int.parse(stdin.readLineSync());
  stdout.writeln('Digite el número de columnas: ');
  ncols = int.parse(stdin.readLineSync());

  //Matriz m = new Matriz(nfilas, ncols);

  for (int i = 0; i < nfilas; i++) {
    for (int j = 0; j < ncols; j++) {
      stdout.writeln('[$i][$j] Digite un número: ');
      val = int.parse(stdin.readLineSync());
      fila.insert(j, val);
    }
    m.insert(i, new List<int>.from(fila));
    fila.clear();
  }

}

void comprobarSimetria(List<List> m) {
  int cont = 0;

  if (nfilas == ncols) {
    for (int i = 0; i < nfilas; i++) {
      for (int j = 0; j < ncols; j++) {
        if (m[i][j] == m[j][i]) {
          cont++;
        }
      }
    }
  }

  if (cont == nfilas * ncols) {
    print('\nLa matriz es Simétrica!\n');
  } else {
    print('\nLa matriz NO es Simétrica!');
  }
}

void main() {

  pedirDatos(matriz);

  print(matriz);

  comprobarSimetria(matriz);

}